src/go/configgenerator/route_generator.go

```diff
@@ -309,22 +318,23 @@ func makeRouteCors(serviceInfo *configinfo.ServiceInfo) (*corspb.CorsPolicy, []*
 	return cors, corsRoutes, nil
 }
 
-// makePerRouteFilterConfig generates the per-route config across all filters for a single method and http rule.
-func makePerRouteFilterConfig(method *configinfo.MethodInfo, httpRule *httppattern.Pattern, filterGenerators []filtergen.FilterGenerator) (map[string]*anypb.Any, error) {
+// makePerResourceFilterConfig generates filter config per-route or per-vHost.
+// Caller is responsible for passing the callback for the corresponding resource.
+func makePerResourceFilterConfig(filterGenerators []filtergen.FilterGenerator, callback perResourceCallback) (map[string]*anypb.Any, error) {
```

Comment:
NIT: makePerResourceFilterConfig and perResourceCallback are confusing.

Can we define makePerRouteFilterConfig/perRouteCallback and makePerHostFilterConfig/perHostCallback separately?

Reply:
I see. I was trying to reduce code duplication but agreed. Done